refm/api/src/rubygems/version/Gem__Version

```diff
@@ -36,66 +36,139 @@ p versions.sort_by{ |v| Gem::Version.new(v) }
 
 == Singleton Methods
 
+#@since 2.5.0
 --- correct?(version) -> bool
 
-version が正しいバージョンであれば真を返します。そうでなければ偽を返します。
+version が正しいバージョンであれば true を返します。そうでなければ false を返します。
+
+#@samplecode
+p Gem::Version.correct?("9.1")       # => true
+p Gem::Version.correct?("incorrect") # => false
+
+p Gem::Version.correct?(nil)         # => true
+# nil versions are discouraged and will be deprecated in Rubygems 4
+# version が nil のときは true を返しますが、推奨はされず、Ruby 2.6以降では警告がでます。
+#@end
+#@else
+--- correct?(version) -> 0 | nil
+
+version が正しいバージョンであれば 0 を返します。そうでなければ nil を返します。
+
+#@samplecode
+p Gem::Version.correct?("9.1")       # => 0
+p Gem::Version.correct?("incorrect") # => nil
+
+p Gem::Version.correct?(nil)         # => 0
+#@end
+#@end
 
 @param version バージョンを文字列か数値で指定します。
 
+
 --- create(input) -> Gem::Version | nil
 
 [[c:Gem::Version]] のインスタンスを作成するためのファクトリメソッドです。
 
-例:
-
-  ver1 = Gem::Version.create('1.3.17')   # => #<Gem::Version "1.3.17">
-  ver2 = Gem::Version.create(ver1)       # => #<Gem::Version "1.3.17">
-  ver3 = Gem::Version.create(nil)        # => nil
+#@samplecode
+ver1 = Gem::Version.create('1.3.17')   # => #<Gem::Version "1.3.17">
+ver2 = Gem::Version.create(ver1)       # => #<Gem::Version "1.3.17">
+ver3 = Gem::Version.create(nil)        # => nil
+#@end
 
 @param input [[c:Gem::Version]] のインスタンスか文字列を指定します。
 
 @raise ArgumentError input がバージョンとして不正なオブジェクトである場合に発生します。
 
 @see [[m:Gem::Version.correct?]]
 
+
+--- new(version) -> Gem::Version
+
+バージョンを表す文字列から、Gem::Version インスタンスを作成します。
+
+引数のバージョンを表す文字列とは、 数字かASCII文字の連続であり、ドットで区切られたものです。
+
+#@samplecode
+p Gem::Version.new('1.2.0a') # => #<Gem::Version "1.2.0a">
+
+#@since 2.5.0
```

Comment:
2.4.1 からに見えるので、 since 2.5.0 よりも since 2.4.0 の方が良いかもしれません。

```
%  docker run -it --rm ghcr.io/ruby/all-ruby env ALL_RUBY_SINCE=ruby-2.3 ./all-ruby -e 'p Gem::Version.new(" ")'
ruby-2.3.0          #<Gem::Version "">
...
ruby-2.4.0          #<Gem::Version "">
ruby-2.4.1          #<Gem::Version "0">
...
ruby-3.0.0          #<Gem::Version "0">
```

Reply:
54d4f5f
2.4以降ならもう分岐なくしていいのかと思ったので、分岐がなくす形で対処しました。